src/gurt/misc.c

```diff
@@ -11,12 +11,38 @@
 
 #include <stdarg.h>
 #include <math.h>
+#include <malloc.h>
 #include <string.h>
 #include <gurt/common.h>
 
 /* state buffer for DAOS rand and srand calls, NOT thread safe */
 static struct drand48_data randBuffer = {0};
 
+d_alloc_track_cb_t d_alloc_track_cb;
+d_alloc_track_cb_t d_free_track_cb;
+static __thread void *track_arg;
```

Comment:
I'm a bit confused by this and I wonder if there's a layering violation here?

There is a cb register function that works across the whole program, then each pthread may also provide a handle which would mean that posix thread invokes the callback?  As an interface that's fairly horrible to work with if we want to use it anywhere else in future - and dumping memory stats on exit would be a great thing for the client to do at some point.

Reply:
If we want to use this on the client side, yes, the current per xstream structure needs to be re-worked.

This patch is only for monitoring the memory usage for the server engine.

Reply:
How does this compare with just calling mallinfo?

Reply:
mallinfo will only show the total memory usage of the node, and the patch is for DAOS engine memory usage?

Reply:
mallinfo tells you the current process, I'm not sure if that's what you want.

Reply:
yes, mallinfo will provide the memory information per process. Though the patch is for tracking the memory per xstream. Hmm those these mallinfo could be useful indeed. I probably can export them through system xstream, but should be done through a different patch.